hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestCallerContext.java

```diff
@@ -42,6 +42,37 @@ public void testBuilderAppend() {
         builder.build().getContext());
   }
 
+  @Test
+  public void testBuilderAppendIfAbsent() {
+    Configuration conf = new Configuration();
+    conf.set(HADOOP_CALLER_CONTEXT_SEPARATOR_KEY, "$");
+    CallerContext.Builder builder = new CallerContext.Builder(null, conf);
+    CallerContext context = builder.append("key1", "value1").build();
+    Assert.assertEquals("key1:value1",
+        builder.build().getContext());
+
+    // Append an existed key.
+    builder.appendIfAbsent("key1", "value1");
+    String[] items = context.getContext().split("\\$");
```

Comment:
Sorry, I have one more request.
```suggestion
    String[] items = builder.build().getContext().split("\\$");
```

Reply:
Thanks for your careful review, I will fix it together.